Sources/OpenAISwift/Models/ChatMessage.swift

```diff
@@ -17,6 +17,14 @@ public enum ChatRole: String, Codable {
     case assistant
 }
 
+/// A structure that represents a single chat message delta in a chat conversation.
+public struct ChatMessageDelta: Codable {
```

Comment:
I wonder why a new struct is needed insted of just using `ChatMessqge` ?

Reply:
The data format is slight different, it's using `delta` in `[{"delta":{"role":"assistant"},"index":0,"finish_reason":null}]`
I don't know if it's standard. Someone asked as well https://community.openai.com/t/gpt-4-createchatcompletion-stream-response-format-and-model-switch/112719
``` 
data: {"id":"chatcmpl-6yVTvD6UAXsE9uG2SmW4Tc2iuFnnT","object":"chat.completion.chunk","created":1679878715,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"role":"assistant"},"index":0,"finish_reason":null}]}
data: {"id":"chatcmpl-6yVTvD6UAXsE9uG2SmW4Tc2iuFnnT","object":"chat.completion.chunk","created":1679878715,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"Once"},"index":0,"finish_reason":null}]}
```

Reply:
In your code, `ChatMessageDelta` and the existing `ChatMessage` are identical. So there is no need to have both, as it is just confusing. 

The difference is handled in the new `Payload` type you introduced, `StreamMessageResult`, where the `delta` field is dealt with.

Reply:
yep, I thought you are talking about `StreamMessageResult`
updated https://github.com/adamrushy/OpenAISwift/pull/63/commits/97111b174cde1494e852df92435e735568bd2ce8